### ISSUE TITLE
[RFE] Look for options to restructure/remove `jobConfig` from indexer docs

### ISSUE THREAD

**Issue by vishnuchalla** [kube-burner/kube-burner#606]
**Is your feature request related to a problem? Please describe.**
Based on our analysis [here](https://docs.google.com/document/d/1gJaS2i4a1d9FItuseqb6X4HvLVEzGKPxcHuurgZPy-Y/edit#heading=h.6ke30kd33a97), We have noticed that `jobConfig` is being published redundantly in every single ES doc that is being indexed. This operation is redundant is not that useful.

**kube-burner regular mode**
```
kube-burner-ocp cluster-density-v2 --iterations=5 --profile-type=regular --user-metadata=user-metadata.yml --uuid=vchalla-testfeb27regular --local-indexing --alerting=true --qps=20 ooburst=20 --churn-cycles=1
```
This command ran cdv2 with regular metrics profile and spit out 5128 json docs in the output constituting **5963 KB** of data (i.e which includes clusterMetadata, jobSummary, metrics, alerts and measurements). One thing to notice here is each document that had a json field called jobConfig which is 464B in size in total adds up to (464B * 5128)/1000 = **2380 KB** in the entire data. This field is redundant and constitutes around 40% of data.

**kube-burner reporting mode**

```
kube-burner-ocp cluster-density-v2 --iterations=5 --profile-type=reporting --user-metadata=user-metadata.yml --uuid=vchalla-testfeb27reporting --local-indexing --alerting=true --qps=20 ooburst=20 --churn-cycles=1
```

This command ran cdv2 with regular metrics profile and spit out 942 json docs in the output constituting **1006 KB** of data (i.e which includes clusterMetadata, jobSummary, metrics, alerts and measurements). Again jobConfig which is 464B in size in total adds up to (464B * 942)/1000 = **437 KB** in the entire data. This field is redundant and constitutes around 40% of data.
So in conclusion, it is recommended to use reporting mode everywhere possible and also get rid of jobConfig in every single doc as it is an extra overhead to ES from storage and retrieval perspective. Having only a single jobConfig that too in the jobSummary document should be suffiecient.

**Describe the solution you'd like**
To see if we can remove `jobConfig` from the indexer docs except from the `jobSummary` document. Seems to be practically possible.

**Describe alternatives you've considered**
Since this is a cleanup activity, we need to get rid of this some or the other way. Starting with a small POC should be good enough for us to verify this and we can make further decisions from there.

**Comment by rsevilla87** [comment]
cc: @kube-burner/ocp-perfscale

**Comment by rsevilla87** [comment]
Hey @vishnuchalla , can we close this issue?
